client/state/analytics/reducer.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Internal dependencies
+ */
+import {
+	ANALYTICS_TRACKING_ON,
+} from 'state/action-types';
+
+export const analyticsTracking = ( state = false, { type } ) => {
+	if ( ANALYTICS_TRACKING_ON === type ) {
+		return true;
+	}
+
+	if ( ANALYTICS_TRACKING_ON === type ) {
+		return false;
+	}
+
+	return state;
+};
+
+export default analyticsTracking;
```

Comment:
What do you think about my suggestions [here](https://github.com/Automattic/wp-calypso/pull/12332#discussion_r109759249) and [here](https://github.com/Automattic/wp-calypso/pull/12332#discussion_r109760170)?